pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -233,8 +233,12 @@ func buildFileAccessLogHelper(path string, mesh *meshconfig.MeshConfig) *accessl
 		}
 		fl.AccessLogFormat = &fileaccesslog.FileAccessLog_LogFormat{
 			LogFormat: &core.SubstitutionFormatString{
-				Format: &core.SubstitutionFormatString_TextFormat{
-					TextFormat: formatString,
+				Format: &core.SubstitutionFormatString_TextFormatSource{
+					TextFormatSource: &core.DataSource{
```

Comment:
In general - using new API SGTM..

But why does the old way not work, and the new way does? I would have thought this is just a syntax change?

Reply:
Thanks. It's just a syntax change. I've investigated the code, but however cannot find why was his `accessLogFormat` field not working. I've revised the PR description, and if I can find the issue, I'll open a new PR for the fix.

Reply:
Ah see, so this was found during https://github.com/istio/istio/issues/34466, but does not *fix* that issue, correct?

Reply:
correct